test/spec/directives/focus-me.directive.spec.js

```diff
@@ -0,0 +1,26 @@
+'use strict';
+
+describe('focusMe Directive', function() {
+  var element, scope;
+
+  beforeEach(module('habitrpg'));
+
+  beforeEach(inject(function($rootScope, $compile) {
+    scope = $rootScope.$new();
+
+    element = "<input focus-me></input>";
+
+    element = $compile(element)(scope);
+    scope.$digest();
+  }));
+
+  it('focuses the element when appended to the DOM', function(done) {
+    inject(function($timeout) {
+      element.appendTo(document.body);
+
+      element.on('focus', function() { done() });
+
+      $timeout.flush();
```

Comment:
My only fear is that relies on ngMock making async behavior "synchronous", which leaves it vulnerable should anything change. Happy to refactor the spec, but if we're testing an async method, it really should be with an async test. Thoughts?